### ISSUE TITLE
Dice:  The `ignore_index` 255 is not valid for inputs with 6 classes

### ISSUE THREAD

**Issue by robmarkcole** [Lightning-AI/torchmetrics#2849]
## 🐛 Bug

6 classes, ignore index 255 and 
```python
Dice(
    num_classes=num_classes,
    ignore_index=ignore_index,
    average="micro",
)
```

Error 
```
  - The `ignore_index` 255 is not valid for inputs with 6 classes
 ```
Similar to https://github.com/Lightning-AI/torchmetrics/issues/304

**Comment by Borda** [comment]
is it with the latest version?

**Comment by robmarkcole** [comment]
With `torchmetrics==1.6.0`

**Comment by Borda** [comment]
I do not think this is really a bug, but it could be an interesting improvement...
@robmarkcole could you interested in sending a PR/suggestion?

**Comment by Isalia20** [comment]
@Borda  I can take a look at this, you can assign me if no objections

**Comment by SkafteNicki** [comment]
@Isalia20 sounds good to me

**Comment by SkafteNicki** [comment]
@Isalia20 thanks for confirming. Closing issue since `Dice` in the classification domain is deprecated and will soon be removed. Consider using `F1Score` metric from classification domain or `DiceScore` metric from segmentation domain.
